### DISCUSSION TITLE
Libraries and tw-merge

### DISCUSSION THREAD

**Discussion by felipedeboni** [dcastil/tailwind-merge#425] [Help] [answered]
Hey,

I have the following scenario:

* **Package A:** It extends `tailwind-merge` using `extendTailwindMerge`.
* **App:** it consumes Package A but also needs to extend `tailwind-merge` further. This way, when any component from Package A is called, `twMerge` will be aware of the app-specific classes.

Any way to solve it?

Thanks :)

**Comment by dcastil** [comment, marked as answer]
Hey @felipedeboni! 👋

Package A needs to give the app a way to configure tailwind-merge, e.g. by allowing to hook into the call to `extendTailwindMerge`.

**Reply by felipedeboni** [reply]
Thanks! In this case, I will be monkey patching it :)

**Reply by dcastil** [reply]
Ah yeah, if you don't have access to Package A, it becomes quite complicated. For patching third-party libraries I can recommend https://github.com/ds300/patch-package.

**Reply by felipedeboni** [reply]
I've used this package for years, now with pnpm I can patch it directly. Works like a charm :D
